src/value/owned.rs

```diff
@@ -1,186 +1,183 @@
 // Copyright Amazon.com, Inc. or its affiliates.
 
-//! Provides owned implementations of [`IonSymbolToken`], [`IonElement`] and its dependents.
-//!
-//! This API is simpler to manage with respect to borrowing lifetimes, but requires full
-//! ownership of data to do so.
-
 use crate::ion_eq::IonEq;
 use crate::text::text_formatter::IonValueFormatter;
 use crate::types::decimal::Decimal;
 use crate::types::integer::Integer;
 use crate::types::timestamp::Timestamp;
-use crate::types::SymbolId;
 use crate::{IonType, Symbol};
 use num_bigint::BigInt;
 use std::collections::HashMap;
 use std::fmt::{Display, Formatter};
 use std::iter::FromIterator;
-use std::rc::Rc;
 
 use crate::symbol_ref::AsSymbolRef;
+use crate::value::builders::{ListBuilder, SExpBuilder, StructBuilder};
 use crate::value::iterators::{
     ElementsIterator, FieldIterator, FieldValuesIterator, IndexVec, SymbolsIterator,
 };
 
-impl Symbol {
-    pub fn symbol_id(&self) -> Option<SymbolId> {
-        match self.text() {
-            Some(_) => None,
-            None => Some(0),
-        }
+impl Element {
+    pub fn null(null_type: IonType) -> Element {
+        null_type.into()
     }
```

Comment:
🗺️ I renamed all of the `new_TYPE` methods to simply `TYPE` (e.g. `Element::new_blob(...)` to `Element::blob(...)` with the exception of the container types. `struct` is a Rust keyword, so the container types' methods' names now indicate that they return the appropriate builder. (e.g. `new_struct()` -> `struct_builder()`)